pkg/imgpkg/bundle/bundle.go

```diff
@@ -58,6 +58,15 @@ func (o *Bundle) addImageRefs(refs ...ImageRef) {
 
 func (o *Bundle) imageRef(imageDigest string) (ImageRef, bool) {
 	ref, found := o.imagesRef[imageDigest]
+	if !found {
+		for _, imgRef := range o.imagesRef {
+			for _, loc := range imgRef.Locations() {
```

Comment:
hmm, i dont think i understand how locations become involved here. afaik everywhere the primary key is `image` field value. are we screwing that up somewhere and that's why we need to start looking at locations?

Reply:
discussed over zoom. short version is that for optimization reasons the imageDigest sometimes references the image ref of the target repo. However the image ref in the images lock (belonging to the bundle) has the original source reference.

Thus, since they are sometimes different they don't match in this function and never make it to the Locations image.

The proposed fix is to iterate over all the locations to try and 'map' an image to a bundle's image ref to decide whether to include it as part of the location OCI.